### PR TITLE
tpm2_eventlog_yaml: fix parsing for MokListTrusted

### LINKED ISSUE #3050: tpm2_eventlog: error Cannot allocate memory when using --eventlog-version=2
I'm having this issue while running it on a docker container with fedora's latest version. 

docker pull fedora
inside the container installed tpm2-tools as follows: sudo dnf install tpm2-tools.x86_64

I copied a binary_bios_measurements file from another machine to the container to perform the test, since it already failed in RHEL8, and run the following command:
tpm2_eventlog --eventlog-version=2 ./binary_bios_measurements

The first part of the output seems to be okay but at the end I see:

ERROR: Failled to allocate data: Cannot allocate memory

ERROR: failed to parse tpm2 eventlog
ERROR: Unable to run tpm2_eventlog

I have the complete output if need, I executed it on a VM so I think it shouldn't be security sensitive.  Let me know if it's useful and I can include it in the issue. 

tpm2_eventlog -v
tool="tpm2_eventlog" version="5.2" tctis="libtss2-tctildr" tcti-default=tcti-device

### PR DESCRIPTION
**Pull request by THS-on**
Not all data in events of the EV_EFI_VARIABLE_AUTHORITY are
EFI_SIGNATURE_DATA. The entry for MokListTrusted is a boolean
encoded as an integer similar to SecureBoot variable.

Fixes #3050

### CODE REVIEW

**Comment by idesai** [comment]
@THS-on please can you add the eventlog binary to the integration test fixtures and update the test.

File: lib/tpm2_eventlog_yaml.c [hunk 1]
```diff
@@ -418,27 +418,49 @@ static bool yaml_uefi_var(UEFI_VARIABLE_DATA *data, size_t size, UINT32 type,
...
-            
-            EFI_SIGNATURE_DATA *s= (EFI_SIGNATURE_DATA *)&data->UnicodeName[
-                data->UnicodeNameLength];
-            char *sdata = calloc (1,
-                BYTES_TO_HEX_STRING_SIZE(data->VariableDataLength - sizeof(EFI_GUID)));
-            if (sdata == NULL) {
-                LOG_ERR("Failled to allocate data: %s\n", strerror(errno));
-                return false;
+            /* The MokListTrusted is boolean option, not a EFI_SIGNATURE_DATA*/
+            if ((strlen(ret) == 14 && strncmp(ret, "MokListTrusted", 10) == 0)) {
```
Comment (idesai) [inline lib/tpm2_eventlog_yaml.c:423]: @THS-on can you please #define the magic numbers here

Comment (THS-on) [reply lib/tpm2_eventlog_yaml.c]: @idesai I can do that but the checks in line 324f for `EV_EFI_VARIABLE_DRIVER_CONFIG` also don't do that.
Should I add defines then also for those in an extra commit?

Comment (idesai) [reply lib/tpm2_eventlog_yaml.c]: Yes, Thank you.

Comment (THS-on) [reply lib/tpm2_eventlog_yaml.c]: Done

**Comment by idesai** [comment]
@JuergenReppSIT, it appears all the FAPI tests are failing with master tpm2-tss branch.

**Comment by JuergenReppSIT** [comment]
> @JuergenReppSIT, it appears all the FAPI tests are failing with master tpm2-tss branch.

I have create a PR which should fix the problem: https://github.com/tpm2-software/tpm2-tss/pull/2434

**Review by idesai — APPROVED** [review]

**Pull request merged**